core/controllers/base.py

```diff
@@ -164,9 +165,18 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
             self.payload = None
         self.iframed = False
 
-        auth_claims = auth_services.get_auth_claims_from_request(request)
-        self.current_user_is_super_admin = (
-            auth_claims is not None and auth_claims.role_is_super_admin)
+        try:
+            auth_claims = auth_services.get_auth_claims_from_request(request)
+        except auth_domain.StaleAuthSessionError:
+            auth_services.destroy_auth_session(self.response)
+            raise self.UnauthorizedUserException('Please sign in again')
+        except auth_domain.InvalidAuthSessionError:
```

Comment:
This scenario should probably also be covered in tests to ensure the correct error code is being raised.

Reply:
Done.